students/perrymant/code-examples/cache/src/CacheTest.java

```diff
@@ -0,0 +1,40 @@
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+/*
+can we create tests to ensure we're not breaking anything please?
+This should give you some practice in dependency injection too,
+as you'll have to give the class a DB implementation when testing.
+*/
+@RunWith(MockitoJUnitRunner.class)
+public class CacheTest {
+
+    @Mock
+    DB db;
+
+    @InjectMocks
+    Cache cache;
+
+    @Test
+    public void canReturnCorrectSizeOf_getAllUsers() {
+        Account account = new Account();
+        account.setUserIds(new HashSet<>(Arrays.asList(11, 22, 33)));
+        Assert.assertEquals(3, cache.getAllUsers(account).size());
```

Comment:
Let's make these tests even more naive. Not 3 items going in, but perhaps just one.
Then try calling it twice, verifying that the DB was called only once.